### PR TITLE
fix(cache):  clean pending request immediately when future cancel

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

close https://github.com/singularity-data/risingwave/issues/3912

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by Little-Wallace** [comment]
@skyzh Please take a look.
But I do not think it is necessary to cancel all pending request immediately because we run `fetch_value` in tokio-pool, it means that this job must give a result.

File: src/common/src/cache.rs [hunk 1]
```diff
@@ -1235,4 +1262,53 @@ mod tests {
         drop(cache);
         assert!(listener.released.lock().is_empty());
     }
+
+    pub struct SyncPointFuture<F: Future> {
+        inner: F,
+        polled: Arc<AtomicBool>,
+        // recv: Receiver<()>,
```
Comment (skyzh) [inline src/common/src/cache.rs:1269]: Remove?

**Review by skyzh — APPROVED** [review]
LGTM, but I'd prefer the [shared future](https://docs.rs/futures/latest/futures/future/struct.Shared.html) approach. In that way we don't need to spawn a new future.

File: src/common/src/cache.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/common/src/cache.rs]: fixed

**Pull request merged**